code/modules/mining/equipment/resonator.dm

```diff
@@ -91,29 +126,42 @@
 	resonance_damage *= damage_multiplier
 
 /obj/effect/temp_visual/resonance/proc/burst()
-	var/turf/T = get_turf(src)
-	new /obj/effect/temp_visual/resonance_crush(T)
-	if(ismineralturf(T))
-		if(is_ancient_rock(T))
-			visible_message("<span class='notice'>This rock appears to be resistant to all mining tools except pickaxes!</span>")
-		else
-			var/turf/simulated/mineral/M = T
-			M.gets_drilled(creator)
-	check_pressure(T)
-	playsound(T,'sound/weapons/resonator_blast.ogg',50,1)
-	for(var/mob/living/L in T)
+	SIGNAL_HANDLER
+	if(rupturing)
+		return
+	rupturing = TRUE
+	var/turf/src_turf = get_turf(src)
+	new /obj/effect/temp_visual/resonance_crush(src_turf)
+	if(ismineralturf(src_turf))
+		var/turf/simulated/mineral/mineral_turf = src_turf
+		mineral_turf.gets_drilled(creator)
+	check_pressure(src_turf)
+	playsound(src_turf, 'sound/weapons/resonator_blast.ogg', 50, TRUE)
+	for(var/mob/living/attacked_living in src_turf)
 		if(creator)
-			add_attack_logs(creator, L, "Resonance field'ed")
-		to_chat(L, "<span class='userdanger'>[src] ruptured with you in it!</span>")
-		L.apply_damage(resonance_damage, BRUTE)
+			log_attack(creator, attacked_living, "used a resonator field on", "resonator")
+			SEND_SIGNAL(creator, COMSIG_LIVING_RESONATOR_BURST, creator, attacked_living)
```

Comment:
If this signal has no listeners, we don't need it being sent. Please remove it for now.

Reply:
yeah, it use for style proc, but i actually didnt find one. Donezo